setup.py

```diff
@@ -632,21 +632,29 @@ def get_common_options(build_ext):
     if not have_gloo and not have_mpi:
         raise RuntimeError('One of Gloo or MPI are required for Horovod to run. Check the logs above for more info.')
 
-    gpu_allreduce = os.environ.get('HOROVOD_GPU_ALLREDUCE')
-    if gpu_allreduce and gpu_allreduce != 'MPI' and gpu_allreduce != 'NCCL' and \
-        gpu_allreduce != 'DDL':
-        raise DistutilsError('HOROVOD_GPU_ALLREDUCE=%s is invalid, supported '
-                             'values are "", "MPI", "NCCL", "DDL".' % gpu_allreduce)
-
-    gpu_allgather = os.environ.get('HOROVOD_GPU_ALLGATHER')
-    if gpu_allgather and gpu_allgather != 'MPI' and gpu_allgather != 'NCCL':
-        raise DistutilsError('HOROVOD_GPU_ALLGATHER=%s is invalid, supported '
-                             'values are "", "MPI", "NCCL".' % gpu_allgather)
-
-    gpu_broadcast = os.environ.get('HOROVOD_GPU_BROADCAST')
-    if gpu_broadcast and gpu_broadcast != 'MPI' and gpu_broadcast != 'NCCL':
-        raise DistutilsError('HOROVOD_GPU_BROADCAST=%s is invalid, supported '
-                             'values are "", "MPI", "NCCL".' % gpu_broadcast)
+    gpu_operations = os.environ.get('HOROVOD_GPU_OPERATIONS')
+    if gpu_operations and gpu_operations not in {'NCCL', 'MPI'}:
+        raise DistutilsError(f'HOROVOD_GPU_OPERATIONS={gpu_operations} is invalid, '
+                             f'supported values are: {{NCCL, MPI}}.')
+
+    def get_gpu_op_variable(op_variable_name, options):
+        gpu_operation = os.environ.get(op_variable_name)
+        if gpu_operations:
+            if gpu_operation:
+                raise DistutilsError(f'Cannot specify both HOROVOD_GPU_OPERATIONS and {op_variable_name} options. '
+                                     'Try unsetting one of these variables and reinstalling.')
+            return gpu_operations
+
+        if gpu_operations and gpu_operation not in options:
+            options_str = ', '.join(options)
+            raise DistutilsError(f'{op_variable_name}={gpu_operation} is invalid, '
+                                 f'supported values are: {{{options_str}}}.')
```

Comment:
why braces in `supported values are: {NCCL, MPI}`?

Reply:
In my mind I was thinking set notation, but removed since it's unclear.